### PR TITLE
Refactors pandemic machine

### LINKED ISSUE #68504: pandemic doesn't html encode names properly
image

### PR DESCRIPTION
**Pull request by jlsnow301**
## About The Pull Request
yes, I did not need to refactor it to fix the problem, but that does not mean it didn't need a refactor

- fixes #68504 
- documents + organizes pandemic code.
- sets pandemic to autoupdate(false) so it should take up less cpu

other notes: my typescript skills were inefficient when I first wrote this ui. I used a spaghetti-on-the-wall approach that could be greatly improved upon if anyone wanted to be thorough at a later date

## Changelog

:cl:
fix: Custom virus names should now display properly in the pandemic.
/:cl:

### CODE REVIEW

File: tgui/packages/common/string.js [hunk 1]
```diff
@@ -175,7 +175,7 @@ export const decodeHtmlEntities = (str) => {
     // Basic entities
     .replace(translate_re, (match, entity) => translate[entity])
     // Decimal entities
-    .replace(/&#?([0-9]+);/gi, (match, numStr) => {
+    .replace(/&#?([0-9]+);?/gi, (match, numStr) => {
```
Comment (jlsnow301) [inline tgui/packages/common/string.js:178]: this was an issue: decodeHtmlEntities was not registering "&#39", which is the apostrophe

**Comment by Mothblocks** [comment]
btw im pretty sure when i looked at the code i saw `sanitize_name(html_encode`, when sanitize already does html encode

File: tgui/packages/common/string.js (hunk 1 above)
Comment (stylemistake) [inline tgui/packages/common/string.js:178]: Semicolon is required for an html entity to be valid, otherwise it's treated as text.

Comment (stylemistake) [reply tgui/packages/common/string.js:178]: https://www.w3.org/TR/html4/sgml/entities.html

**Comment by timothymtorres** [comment]
Very nice!

File: tgui/packages/common/string.js (hunk 1 above)
Comment (jlsnow301) [reply tgui/packages/common/string.js:178]: Good, I'll revert this. I felt hesitant asking this conditional. The problem seems to be in the extra layer of sanitization as mentioned by @Mothblocks

**Review by stylemistake — APPROVED** [review]

**Comment by jlsnow301** [comment]
this is ready whenever

**Pull request merged**